tests/integration/models/reddit/test_live.py

```diff
@@ -42,3 +46,115 @@ def test_updates(self, _):
             for update in thread.updates(limit=None):
                 assert update.thread == thread
         assert update.body.startswith('Small change:')
+
+
+class TestLiveContributorRelationship(IntegrationTest):
+    @contextmanager
+    @mock.patch('time.sleep', return_value=None)
+    def ensure_no_invite(self, thread, redditor, _):
+        thread.contributor.remove_invite(redditor)
+        yield
+        thread.contributor.remove_invite(redditor)
```

Comment:
I know the cassettes are already generated, but can we do without this context manager, or at least only doing the before-state clean up? I did a lot of this in earlier versions of PRAW. What I came to dislike about the approach is it makes it less clear to see what the test is testing for when looking at the cassette as there are more requests than necessary.

My approach typically when generating cassettes is to manually set the state to what I expect it to be prior to running the test for the first time. While this means the cassettes cannot be easily generated in bulk, it means the test code and the cassettes are simpler.